src/blueapi/service/main.py

```diff
@@ -304,6 +304,7 @@ def get_tasks(
 @start_as_current_span(TRACER, "task.task_id")
 def set_active_task(
     task: WorkerTask,
+    instrument_session: str,
```

Comment:
Could: The approach here is to make tasks independent of an instrument session and only pass the session when stating them. The alternative is to embed the session in the task, which I think I'm more in favour of. The idea of this API is that it is transaction style where tasks are the transaction (i.e. cheap and binnable and everything about them is set up in advance) and `set_active_task` just commits the transaction. I think I would prefer the session embedded in the task, but open to discussing.

Reply:
We check authorization to write to a visit as part of beginning a task, as part of the call to the NumTracker, so I think the instrument_session should be passed as part of the request to begin the task. 

See also discussion on #774: if user_a with permission to write to x,y creates a task and user_b with permission to write to x submits it- if user_a has incorrectly set the instrument_session to y, user_b cannot run the scan due to permissions issues. 

Or if user_b maliciously creates a task with instrument_session y to allow their data to be written to e.g. the commissioning directory and so not archived or exposed.

Submitting the task to be run is the user knowingly saying "collect <this data>", and I think that should include "to <this instrument_session>"

Reply:
Per offline discussion: This shouldn't be merged until we do an AuthZ check of session/user, but that doesn't actually affect this dicussion. The session is embedded in the task and is checked against the user who triggers the task. The question of whether there should be separate AuthZ to check that the creator and submitter are the same user doesn't affect this.